openid-connect-server/src/test/java/org/mitre/oauth2/service/impl/TestDefaultOAuth2AuthorizationCodeService

```diff
@@ -0,0 +1,129 @@
+package org.mitre.oauth2.service.impl;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+import java.util.Date;
+import java.util.HashMap;
+
+import org.easymock.Mock;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mitre.oauth2.model.AuthenticationHolderEntity;
+import org.mitre.oauth2.model.AuthorizationCodeEntity;
+import org.mitre.oauth2.model.ClientDetailsEntity;
+import org.mitre.oauth2.model.DeviceCode;
+import org.mitre.oauth2.repository.AuthenticationHolderRepository;
+import org.mitre.oauth2.repository.AuthorizationCodeRepository;
+import org.mitre.oauth2.repository.impl.JpaAuthenticationHolderRepository;
+import org.mockito.InjectMocks;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.common.util.RandomValueStringGenerator;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Request;
+
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestDefaultOAuth2AuthorizationCodeService {
+	
+	
+	private String codeNotFound;
+	
+	private AuthenticationHolderEntity authHolder;
+	
+	private AuthorizationCodeEntity entity;
+	
+	//are these 3 mocks important?
+	@Mock
+	private AuthorizationCodeRepository repository;
+	
+	@Mock
+	private AuthenticationHolderRepository authenticationHolderRepository;
+	
+	
+	//following two mock suggested by justin
+	@Mock
+	private JpaAuthenticationHolderRepository  authRepo;
```

Comment:
you don't need this mock and the one above it -- use only the first of these two (that uses the interface instead of a concrete class)